### ISSUE TITLE
Generate has wrong year format.

### ISSUE THREAD

**Issue by fmalchemist92** [kurtraschke/pyRFC3339#14]
`<class 'datetime.datetime'> 0999-01-01 00:00:00+00:00` datetime like this will loses leading zero
`999-01-01T00:00:00Z` .

`from datetime import datetime`
`import pyrfc3339`
`assert pyrfc3339.generate(datetime(999, 1, 1), accept_naive=True) == '0999-01-01T00:00:00Z'`

**Comment by kurtraschke** [comment]
I cannot reproduce this under any currently supported Python version.

**Comment by kurtraschke** [comment]
Never mind, it's failing in CI, which suggests it's OS-specific? Will investigate further.

**Comment by kurtraschke** [comment]
This is apparently a platform bug which was fixed in python/cpython#120713 (and first reported in python/cpython#57514).  I am disinclined to add a [Django-style hack](https://github.com/django/django/blob/0e94f292cda632153f2b3d9a9037eb0141ae9c2e/django/utils/formats.py#L266-L273) given that this is now fixed upstream in Python.

Documenting existing behavior for posterity:

```
Python 3.11.2 (main, Aug 26 2024, 07:20:54) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import datetime
>>> datetime(999,1,1,0,0,0).strftime('%Y-%m-%d')
'999-01-01'
```

```
Python 3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import datetime
>>> datetime(999,1,1,0,0,0).strftime('%Y-%m-%d')
'0999-01-01'
```
